dm/worker/server.go

```diff
@@ -446,24 +457,30 @@ func (s *Server) QueryStatus(ctx context.Context, req *pb.QueryStatusRequest) (*
 // QueryError implements WorkerServer.QueryError
 func (s *Server) QueryError(ctx context.Context, req *pb.QueryErrorRequest) (*pb.QueryErrorResponse, error) {
 	log.L().Info("", zap.String("request", "QueryError"), zap.Stringer("payload", req))
+	sourceStatus := s.getSourceStatus(true)
+	sourceError := pb.SourceError{
+		Source: sourceStatus.Source,
+		Worker: s.cfg.Name,
+	}
+	if sourceStatus.Result != nil && len(sourceStatus.Result.Errors) > 0 {
+		sourceError.SourceError = sourceStatus.Result.Errors[0].String()
```

Comment:
Can we `join` all `Result.Errors` here? or it will have at most one error?

Reply:
addressed in https://github.com/pingcap/dm/pull/490/commits/b9f4793fa1b4d18afa02c9fd9987ce42991dad44